packages/website/src/routes/docs/headless/card/card.css

```diff
@@ -0,0 +1,23 @@
+.custom-card {
```

Comment:
**Question:** When do we decide to add css to headless components.

I was asking this myself for the Dialog-Component since it has a backdrop and I needed to specify width and height of the dialog.

Reply:
I have exact the same doubts.
First I decided to added multiple examples in the headless doc: with or without custom classes: I need to try if the headless component really works fine with no classes, by default) but also that can be easily customized.

Anyway there are some  headless components that don't need styles/classes to work (i.e. a Button or the Card that just to display content ).
In other (complex) use cases just like a dropdown or a dialog I think that we should embed some styles in the component itself otherwise the component doesn't work as expected by default.
In the Dialog, for example, I think we should embed the CSS it needs to display it in overlay (with some parameters like centered, fullscreen or something else to help users to easily use it in the most common scenarios ) but also give to the users the possibility to override them adding custom classes / inline styling to each element of the Dialog (backdrop, the dialog panel, title and so on)

Reply:
This headless flow makes sense 👍

Reply:
I have some thoughts on what "headless" means to the end user.

Too often, when I look at other UI libraries, say Bootstrap, they come with a ton of pre-configured styles/options which tend to be an additional burden for developers to overwrite or introduce extra complexity that they should be aware of. This makes it harder for users to compose their own complex components because they can and likely will get confused why their custom styles or functionality doesn't work the way they expect.

Honestly, I think the only styles that should be added to headless components are the ones that are _strictly_ for functionality/behaviour-related purposes. That also goes for options like variants that are meant to be "convenient" for the user.

For example, in the current headless Card implementation, the `variant` prop is provided only in the Daisy implementation and that's good because as a user, when I import the headless Card component, I'm expecting only semantic markup and structure that I can freely style and then incorporate into my own design system with my own `variant` prop implementation.

I think what users expect for a headless component is for it to be just like a native HTML element, but without the styling restrictions. Libraries like Headless UI, Reach UI and Radix UI have raised the bar and included accessibility out of the box and I think Qwik UI should keep to the expectations that those libraries have set.

Using an analogy, I view headless components like a headless CMS. A headless CMS just provides me the structure of the data I need to bring content to a bespoke front-end. I'd like my headless components to also simply provide me the semantic structure and accessibility behaviours I need to create my own bespoke design system of components.

Reply:
Interesting discussion.  I agree @shiroinegai.

So, if we talk about the dialog component that  @GregOnNet is creating, the headless version should:

- set the dialog  in overlay? or not even that?
( it should set the display property to  fixed/absolute style? and probably its default size?)

- handle the backdrop style?
for example, should it handle a default background color? 

- but I think we are all agree that dialog title, content and footer/actions should just contains the HTML and nothing about styling.


The problem is that it's very hard to imagine the final result for end-users that read the documentation.
And probably it will be very hard to skin for people that has not enough CSS knowledge.

So I think the  key is a good documentation with nice examples:
In fact, if headless components will be totally without styles we should always show one or more examples with custom classes showing the final result, just for inspiration or to copy and quickly update the CSS

Reply:
Looking at the headless Dialog implementation from @GregOnNet , since it returns a native `dialog` element, I think it's best to not include any additional styling. You can see the default browser styles for a `dialog` on MDN and it has a 3px black border with 16px padding.

I think the point with headless components isn't to look nice out of the box, but to behave correctly even though they may look absolutely ugly. Radix UI does a really fantastic job using a `styles.css` tab in their docs. If you import just the component parts, it really comes unstyled. You can see how their Dropdown component looks with just the markup alone as demo-ed by one of their maintainers here: https://youtu.be/pcMYcjtWwVI?t=535

> The problem is that it's very hard to imagine the final result for end-users that read the documentation.
And probably it will be very hard to skin for people that has not enough CSS knowledge.

I think this is why we'll be having the Daisy and Material implementations. Those who aren't keen on deeply customising the look of their components can just import those implementations instead. 

These implementations should be composed with our headless components, just that we are also porting over the design (and perhaps even props) that users of those libraries are familiar with. And especially for Daisy, it will also be an enhanced experience for users because it is now composed of Qwik UI's accessible components.

Familiar look, but now accessible!